pkg/sql/catalog/internal/catkv/catalog_query.go

```diff
@@ -191,9 +191,6 @@ func build(
 	if expectedType != catalog.Any && b.DescriptorType() != expectedType {
 		return nil, pgerror.Newf(pgcode.WrongObjectType, "descriptor is a %s", b.DescriptorType())
 	}
-	if err := b.RunPostDeserializationChanges(); err != nil {
-		return nil, errors.NewAssertionErrorWithWrappedErrf(err, "error during RunPostDeserializationChanges")
```

Comment:
I wonder if we should move this error wrapping to the `RunPostDeserializationChanges ` call in `descbuilder`. I'm less sure about the assertion error but the message wrapping seems useful; however it's not like we encounter these often in the wild either.